target/linux/generic/files/drivers/net/phy/ar8216.c

```diff
@@ -2892,6 +2892,19 @@ static int __init ar8216_init(void)
 {
 	int ret;
 
+	if (of_find_node_with_property(NULL, "qca,disable-legacy-phy")) {
+		pr_warn("ar8xxx: skipping legacy drivers as requested.\n");
+		return -ENODEV;
+	}
+
+	if (of_find_compatible_node(NULL, NULL, "qca,qca8327") ||
+	    of_find_compatible_node(NULL, NULL, "qca,qca8328") ||
+	    of_find_compatible_node(NULL, NULL, "qca,qca8334") ||
```

Comment:
Matching `ar8xxx_mdiodev_of_match` and deleting `switch0@1f` in dts should be better. Something like this:
```
if (!of_find_matching_node(NULL, ar8xxx_mdiodev_of_match))
	return -ENODEV;
-------------------------------------------------------------------
&mdio1 {
/delete-node/ switch0@1f;
};
```

Reply:
ar8xxx_mdiodev_of_match does not contain correct compatible strings that should be blocked.
Removing the switch node does not guarantee that the old phy driver will not be used. The old driver is used if the mdio node has child phy node which id matches the old phy driver.

Reply:
It's not true, this check is before Phy registration.